editor/src/components/canvas/canvas-strategies/absolute-reparent-strategy.spec.tsx

```diff
@@ -186,7 +186,7 @@ function reparentElement(
     [],
     [],
     strategyResult.commands,
-    'permanent',
```

Comment:
A few of these specs have something like this that has changed from `'permanent'` to `'end-interaction'`, should they really be `'mid-interaction'`?

Reply:
Correction, `'always'`, not `'mid-interaction'`.

Reply:
we used the same type for the command and for the foldAndApplyCommand... functions.

in this PR I split the type: 
1. The command object has `WhenToRun = 'mid-interaction' | 'always' | 'on-complete'`, 
2. and the actual command _runner_ functions now have `commandLifecycle: 'mid-interaction' | 'end-interaction'`.

Reply:
Ah, I think them both having `mid-interaction` is the root of my confusion.

Reply:
> Ah, I think them both having `mid-interaction` is the root of my confusion.

should I try to unify the 'on-complete' and 'end-interaction' names somehow? I think 'on-complete' doesn't work as the lifecycle's name....

Reply:
No I think it makes sense as it is, the terms make sense in their respective locations.